code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,305 @@
+/datum/action/xeno_action/activable/web_spit
+	name = "Web Spit"
+	ability_name = "Web Spit"
+	mechanics_text = "We spit a stretchy web at our prey"
+	action_icon_state = "web_spit"
+	plasma_cost = 125
+	cooldown_timer = 10 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_WEB_SPIT
+
+/datum/action/xeno_action/activable/web_spit/use_ability(atom/target)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/datum/ammo/xeno/web/web_spit = GLOB.ammo_list[/datum/ammo/xeno/web]
+	var/obj/projectile/newspit = new /obj/projectile(get_turf(X))
+
+	newspit.generate_bullet(web_spit, web_spit.damage * SPIT_UPGRADE_BONUS(X))
+	newspit.def_zone = X.get_limbzone_target()
+
+	newspit.fire_at(target, X, null, newspit.ammo.max_range)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Leash Ball
+// ***************************************
+
+/datum/action/xeno_action/activable/leash_ball
+	name = "Leash Ball"
+	ability_name = "Leash Ball"
+	mechanics_text = " Spit a huge ball of web that snares groups of marines "
+	action_icon_state = "leash_ball"
+	plasma_cost = 250
+	cooldown_timer = 20 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_LEASH_BALL
+
+/datum/action/xeno_action/activable/leash_ball/use_ability(atom/A)
+	var/turf/target = get_turf(A)
+	var/mob/living/carbon/xenomorph/X = owner
+	X.face_atom(target)
+	if(!do_after(X, 1 SECONDS, TRUE, X, BUSY_ICON_DANGER))
+		return fail_activate()
+
+	var/datum/ammo/xeno/leash_ball = GLOB.ammo_list[/datum/ammo/xeno/leash_ball]
+	var/obj/projectile/newspit = new /obj/projectile(get_turf(X))
+
+	newspit.generate_bullet(leash_ball)
+	newspit.fire_at(target, X, null, newspit.ammo.max_range)
+	succeed_activate()
+	add_cooldown()
+
+/obj/structure/xeno/aoe_leash
+	name = "Snaring Web"
+	icon = 'icons/Xeno/Effects.dmi'
+	icon_state = "aoe_leash"
+	desc = "Looks very sticky"
+	destroy_sound = "alien_resin_break"
+	max_integrity = 960
+	obj_integrity = 0
+	layer = ABOVE_ALL_MOB_LAYER
+	anchored = TRUE
+	throwpass = FALSE
+	density = TRUE
+	/// Radius for how far the leash should affect humans and how far away they may walk
+	var/leash_radius = 5
+	/// How much more integrity aoe_leash gains per caught marine, it is preferable that max_integrity is this var * 8.
```

Comment:
This is essentially adding a bit more than half of a resin wall per marine, and with the high max_integrity, it would be stronger than several resin walls. Unless my mathematical assumptions are incorrect, I highly recommend ensuring that the snaring web is not stronger than 2 resin walls.

Reply:
Due to the armor present on resin walls, each 120 health increase per marine only accounts to 0.16 resin walls, reaching its maximum at 8 marines.
A better comparison would be a bit more than one resin door per marine, which I am sure you know how fragile those are.